mono/mini/interp/interp.c

```diff
@@ -99,14 +100,179 @@ typedef struct {
 	InterpFrame *base_frame;
 } FrameClauseArgs;
 
+/*
+ * This code synchronizes with interp_mark_stack () using compiler memory barriers.
+ */
+
+static StackFragment*
+stack_frag_new (int size)
+{
+	StackFragment *frag = (StackFragment*)g_malloc (size);
+
+	frag->pos = (guint8*)&frag->data;
+	frag->end = (guint8*)frag + size;
+	frag->next = NULL;
+	return frag;
+}
+
 static void
-init_frame (InterpFrame *frame, InterpFrame *parent_frame, InterpMethod *rmethod, stackval *method_args, stackval *method_retval)
+frame_stack_init (FrameStack *stack, int size)
+{
+	StackFragment *frag;
+
+	frag = stack_frag_new (size);
+	stack->first = stack->last = stack->current = frag;
+	mono_compiler_barrier ();
+	stack->inited = 1;
+}
+
+static StackFragment*
+add_frag (FrameStack *stack, int size)
+{
+	StackFragment *new_frag;
+
+	// FIXME:
+	int frag_size = 4096;
```

Comment:
Since the default thread stack size on linux is 8mb, I think 4k is kinda small.  My guess would be something like 256k or 512k would be a good start, but really we should have an environment variable or interp flag to control the chunk size and do some experiments. (I assume that is what you meant with `FIXME` 😛 )

Reply:
maybe not quite 256k but I agree it should be increased. 4k is only 10-20calls.